packages/wrangler/src/type-generation/runtime/index.ts

```diff
@@ -0,0 +1,100 @@
+import { readFileSync } from "fs";
+import { writeFile } from "fs/promises";
+import { Miniflare } from "miniflare";
+import { ensureDirectoryExists } from "../../utils/log-file";
+import type { Config } from "../../config/config";
+
+const DEFAULT_OUTFILE_RELATIVE_PATH = "./.wrangler/types/runtime.d.ts";
+
+/**
+ * Generates runtime types for a Workers project based on the provided project configuration.
+ *
+ * This function is designed to be isolated and portable, making it easy to integrate into various
+ * build processes or development workflows. It handles the whole process of generating runtime
+ * types, from ensuring the output directory exists to spawning the workerd process (via Miniflare)
+ * and writing the generated types to a file.
+ *
+ * @throws {Error} If the config file does not have a compatibility date.
+ *
+ * @example
+ * import { generateRuntimeTypes } from './path/to/this/file';
+ * import { readConfig } from './path/to/config';
+ *
+ * const configPath = './wrangler.toml';
+ * const config = readConfig(configPath);
+ * const outFile = '/Users/me/my-project/dist/runtime.d.ts'
+ *
+ * // This will generate runtime types and write them to ./.wrangler/types/runtime.d.ts
+ * await generateRuntimeTypes({ config });
+ *
+ * * // This will generate runtime types and write them to /Users/me/my-project/dist/runtime.d.ts
+ * await generateRuntimeTypes({ config, outFile });
+ *
+ * @remarks
+ * - The generated types are written to a file specified by DEFAULT_OUTFILE_RELATIVE_PATH.
+ * - This could be improved by hashing the compat date and flags to avoid unnecessary regeneration.
+ */
+export async function generateRuntimeTypes({
+	config,
+	generateFn = generate,
+	outFile = DEFAULT_OUTFILE_RELATIVE_PATH,
+}: {
+	config: Config;
+	generateFn?: typeof generate;
```

Comment:
It doesn't look like this option is used by callers of this function—what's the purpose of it?

Reply:
Good spot - that was left in from previous testing.